File: in_Java/0557 Reverse Words in a String III.java
class Solution {
    public String reverseWords(String s) {
        StringBuilder sb = new StringBuilder();
        String[] ss = s.split(" ");
        for(String x : ss){
            sb.append(rev(x));
            sb.append(" ");
        }
        return sb.toString().strip();
    }
    private String rev(String s){
        char[] ch = s.toCharArray();
        int i = 0;
        int j = ch.length-1;
        char t;
        while(i < j){
            t = ch[i];
            ch[i] = ch[j];
            ch[j] = t;
            i++;
            j--;
        }
        return String.valueOf(ch);
    }
}

File: in_Python/0557 Reverse Words in a String III.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        ls = s.split(' ')
        st = ""
        for ss in ls:
            st += ss[::-1] + ' '
        return st[0:len(st)-1]
        